samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -0,0 +1,515 @@
+		LOG_INF("MFP: %s", wifi_mfp_txt(status.mfp));
+		LOG_INF("RSSI: %d", status.rssi);
+		LOG_INF("TWT: %s", status.twt_capable ? "Supported" : "Not supported");
+	}
+	return 0;
+}
+
+static void handle_wifi_connect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *) cb->info;
+
+	if (context.connected) {
+		return;
+	}
+
+	if (status->status) {
+		LOG_ERR("Connection failed (%d)", status->status);
+	} else {
+		LOG_INF("Connected");
+		context.connected = true;
+	}
+
+	context.connect_result = true;
+}
+
+static void handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status =
+		(const struct wifi_status *) cb->info;
+
+	if (!context.connected) {
+		return;
+	}
+
+	if (context.disconnect_requested) {
+		LOG_INF("Disconnection request %s (%d)",
+			 status->status ? "failed" : "done",
+					status->status);
+		context.disconnect_requested = false;
+	} else {
+		LOG_INF("Received Disconnected");
+		context.connected = false;
+	}
+
+	cmd_wifi_status();
+}
+
+static void wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				     uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		handle_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		handle_wifi_disconnect_result(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+static void print_dhcp_ip(struct net_mgmt_event_callback *cb)
+{
+	/* Get DHCP info from struct net_if_dhcpv4 and print */
+	const struct net_if_dhcpv4 *dhcpv4 = cb->info;
+	const struct in_addr *addr = &dhcpv4->requested_ip;
+	char dhcp_info[128];
+
+	net_addr_ntop(AF_INET, addr, dhcp_info, sizeof(dhcp_info));
+
+	LOG_INF("DHCP IP address: %s", dhcp_info);
+}
+
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+				    uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+	case NET_EVENT_IPV4_DHCP_BOUND:
+		print_dhcp_ip(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+#ifdef CONFIG_RAW_TX_PACKET_SAMPLE_CONNECTION_MODE
+static int __wifi_args_to_params(struct wifi_connect_req_params *params)
+{
+	params->timeout = SYS_FOREVER_MS;
+
+	/* SSID */
+	params->ssid = CONFIG_RAW_TX_PACKET_SAMPLE_SSID;
+	params->ssid_length = strlen(params->ssid);
+
+#if defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2)
+	params->security = 1;
+#elif defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA2_256)
+	params->security = 2;
+#elif defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_WPA3)
+	params->security = 3;
+#else
+	params->security = 0;
+#endif
+
+#if !defined(CONFIG_RAW_TX_PACKET_SAMPLE_KEY_MGMT_NONE)
+	params->psk = CONFIG_RAW_TX_PACKET_SAMPLE_PASSWORD;
+	params->psk_length = strlen(params->psk);
+#endif
+	params->channel = WIFI_CHANNEL_ANY;
+
+	/* MFP (optional) */
+	params->mfp = WIFI_MFP_OPTIONAL;
+
+	return 0;
+}
+
+static int wifi_connect(void)
+{
+	struct net_if *iface = net_if_get_default();
```

Comment:
I am aware that this coding style is used in almost all the samples and other zephyr network samples as well. It is a belief that net_if_get_default will always return an iface. But, the function can return NULL. I would always prefer this line be broken up into two parts and code be written as below. One's code should always be able to catch any unintended results in our applications that we write

struct net_if *iface = NULL;
iface = net_if_get_default();
if (iface == NULL) {
     LOG_ERR("iface is NULL");
}

Reply:
We should all transition to `net_if_get_first_wifi()` and of course do a NULL check, just in case.